src/services/mutations/useBookmarkRecipe.ts

```diff
@@ -0,0 +1,21 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { addDocument } from "../firestoreServices";
+import { BookmarkedRecipes } from "../../types/documentTypes";
+
+export const useBookmarkRecipe = () => {
```

Comment:
for the hook, please write return type. When project is getting bigger it's easier to understand what's comming from each hook.

interface UseBookmarkRecipeReturn {
isSaving: boolean,
.....
}